sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/higherOrderFunctions.scala

```diff
@@ -101,6 +101,14 @@ case class NamedLambdaVariable(
   override def simpleString(maxFields: Int): String = {
     s"lambda $name#${exprId.id}: ${dataType.simpleString(maxFields)}"
   }
+
+  // We need to include the Expr ID in the Codegen variable name since several tests bypass
+  // `UnresolvedNamedLambdaVariable.freshVarName`
+  lazy val variableName = s"${name}_${exprId.id}"
```

Comment:
It does not seem consistent with `simpleString`. Is this intentional?

Reply:
Hmm must have thought it was used in the actual generated code or something but doesn't look like it is. In fact I don't think `variableName` is really needed at all, can just use `exprId.id` as the map key, and `name` for the code comment